core/modules/user/js/user.js

```diff
@@ -166,9 +174,28 @@ Backdrop.evaluatePasswordStrength = function (password, username) {
   // its length. Again, adapted from zxcvbn.
   strength = (Math.log(cardinality) / Math.log(2)) * password.length + 1;
 
-  // Check if password is the same as the username.
-  if (password !== '' && password.toLowerCase() === username.toLowerCase()) {
-    strength = 5;
+  // Check if password is the same as the username or email.
+  if (password !== '') {
+    if (password === username || password === email) {
+      strength = 5;
+    }
+    // Consider admin password constraint settings if active.
+    if (config.constraints_active === "1") {
```

Comment:
I'm concerned about variable types when checking the config. The update hook casts them to boolean but here the code checks for numeric and strings. I don't know if we can be certain of this.

Reply:
I think that this is bad DX in general @herbdool ...please see my earlier comment re that: https://github.com/backdrop/backdrop/pull/3053/files#r461499681